include/onnxruntime/core/graph/graph.h

```diff
@@ -808,6 +808,11 @@ class Graph {
 
   virtual ~Graph();
 
+  /** Returns true if the inputs to the graph proto contains FP16 datatype */
+  bool InputsUseFP16() {
+    return inputsHaveFp16_;
```

Comment:
is there a reason why the value is cached instead of just created on demand?

Reply:
It is read at a later time.

---
In reply to: [382312044](https://github.com/microsoft/onnxruntime/pull/3061#discussion_r382312044) [](ancestors = 382312044)

Reply:
right. what I'm saying is why not calculate it at the later time instead of in the initialization? as far as I can tell, there's only one time the value is read so nothing is saved by caching it.

---
In reply to: [382313002](https://github.com/microsoft/onnxruntime/pull/3061#discussion_r382313002) [](ancestors = 382313002,382312044)

Reply:
Right now checking for fp16 in graphproto is called once for telemetry purposes. If there is another check for fp16 in the future, this value is cached and no need for more calculation

---
In reply to: [382322355](https://github.com/microsoft/onnxruntime/pull/3061#discussion_r382322355) [](ancestors = 382322355,382313002,382312044)